### PR TITLE
fix: 1d and 2d plot & title-> label

### PR DESCRIPTION
**Pull request by LovelyBuggies**
- [x] fix: return type of plotting family 
- [x] fix :fig and ax pass in
- [x] docs: make a slight change to allow all names

### CODE REVIEW

**Comment by LovelyBuggies** [comment]
Two issues before merge @henryiii :

- [ ] Should I give ax_res value everytime it draws a different plot (errorbar then plot then show_diff, etc.)
- [ ] There are many assert in the basehist.py to teach mypy the data type. Better if we can remove them and pass.

**Comment by henryiii** [comment]
> 2.0.0 beta1

Let's rename title to label before releasing a beta.

File: src/hist/basehist.py [hunk 1]
```diff
@@ -468,38 +459,57 @@ def plot2d(
 
     def plot2d_full(
         self,
+        *,
         fig: Optional[matplotlib.figure.Figure] = None,
         main_ax: Optional[matplotlib.axes.Axes] = None,
         top_ax: Optional[matplotlib.axes.Axes] = None,
         side_ax: Optional[matplotlib.axes.Axes] = None,
```
Comment (henryiii) [inline src/hist/basehist.py:466]: I thought this was going to be `ax_dict =`

Comment (LovelyBuggies) [reply src/hist/basehist.py:466]: I have implemented in this way hours ago. But abandoned, think about this:

```python
plot2d_full(fig=fig, ax_dict={"main": main_ax, "top": top_ax, "side": side_ax}, ...)
plot2d_full(fig=fig, main_ax=main_ax, top_ax=top_ax, side_ax=side_ax, ...)
```

Which would be a better shortcut for HEPists? I think the second, as they won't require structured manipulation -- encapsulate to a dict (or list, set ...) -- and only directly pass objects in. But if you want them to be better-structured, it's also reasonable.

Comment (henryiii) [reply src/hist/basehist.py:466]: The problem is that your second example doesn't tell me that I need all 3 at the same time. There's no way in Python's signatures to indicate that one keyword argument will require the others. While the dict clearly tells me it it needs several axes. We can even make a TypedDict and MyPy might help a user add the correct keys.

I don't expect users to often make & pass the dict - I expect 90% of users to leave it empty, and many of the remaining 10% can pass `fig=` and still have the axes created for them.

Comment (LovelyBuggies) [reply src/hist/basehist.py:466]: Oh, just to note that, as for the manipulation for the pass-in arguments, we can handle them in both ways.

Comment (henryiii) [reply src/hist/basehist.py:466]: You can always write:

```python
plot2d_full(fig=fig, ax_dict=dict(main_ax=main_ax, top_ax=top_ax, side_ax=side_ax), ...)
```

**Comment by henryiii** [comment]
>  ax_res value everytime it draws a different plot

The result should always be the same. If there's a way to draw a colorbar directly, don't return it - a user who wants it should do it the proper way in matplotlib.

**Comment by LovelyBuggies** [comment]
> The result should always be the same.

So just assign once is okay?

> Let's rename title to label before releasing a beta.

Figure it out in the next PR.

File: src/hist/basehist.py (hunk 1 above)
Comment (LovelyBuggies) [reply src/hist/basehist.py:466]: Okay, will change back to it 😁

Comment (LovelyBuggies) [reply src/hist/basehist.py:466]: Do we also need to give ax_dict to plot1d and plot2d (one axis plotting) to keep the uniform of API?

Comment (henryiii) [reply src/hist/basehist.py]: No, plotting onto multiple axes is different than the (more common) plot to one axis.

**Comment by henryiii** [comment]
I'm going to manually squash all but the last commit, then rebase & merge. :)

**Pull request merged**